### PR TITLE
Update deps

### PR DESCRIPTION
**Pull request by dcoutts**

### CODE REVIEW

**Comment by dcoutts** [comment]
Quite a few changes, need to review carefully.

File: cardano-config/src/Cardano/Config/Shelley/Orphans.hs [hunk 1]
```diff
@@ -82,12 +83,37 @@ instance Crypto crypto => ToJSON (ShelleyGenesis crypto) where
...
+    , sgMaxMajorPV            :: !Natural
+    , sgMaxLovelaceSupply     :: !Word64
+    , sgProtocolParams        :: !SL.PParams
+    , sgGenDelegs             :: !(Map (SL.KeyHash 'SL.Genesis c) (SL.KeyHash 'SL.GenesisDelegate c))
+    , sgInitialFunds          :: !(Map (SL.Addr c) SL.Coin)
+    , sgStaking               :: !(ShelleyGenesisStaking c)
+    }
+  deriving stock    (Eq, Show, Generic)
+  deriving anyclass (NoUnexpectedThunks)
+-}
```
Comment (intricate) [inline cardano-config/src/Cardano/Config/Shelley/Orphans.hs:108]: Should we just remove this then?

**Comment by intricate** [comment]
@dcoutts: Should we make changes to `mkfiles.sh` before merging? That script will be broken following these changes.

File: cardano-config/src/Cardano/Config/Shelley/Orphans.hs (hunk 1 above)
Comment (dcoutts) [reply cardano-config/src/Cardano/Config/Shelley/Orphans.hs:108]: sure

**Comment by dcoutts** [comment]
bors merge

File: cabal.project [hunk 2]
```diff
@@ -302,99 +302,113 @@ source-repository-package
...
   subdir: ouroboros-consensus
 
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/ouroboros-network
-  tag: 9d9754c9ddcfff82b27c371a545aa4680d86d996
-  --sha256: 18ws841jn6hhmm3pqd22lmy20cgnp430dk3s07jzw3d5bpf3i34v
+  tag: 0143c516f860a2c790911bc4cc177569d7d04710
+  --sha256: 0608rashhvybqd1h6fjq9bvnzbyywhxbjxxr9h55m5fjp7i2v3sz
+  subdir: ouroboros-consensus/ouroboros-consensus-test-infra
```
Comment (mrBliss) [inline cabal.project:335]: Why is this needed?

File: cabal.project [hunk 3]
```diff
@@ -302,99 +302,113 @@ source-repository-package
...
   subdir: ouroboros-consensus-shelley
 
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/ouroboros-network
-  tag: 9d9754c9ddcfff82b27c371a545aa4680d86d996
-  --sha256: 18ws841jn6hhmm3pqd22lmy20cgnp430dk3s07jzw3d5bpf3i34v
+  tag: 0143c516f860a2c790911bc4cc177569d7d04710
+  --sha256: 0608rashhvybqd1h6fjq9bvnzbyywhxbjxxr9h55m5fjp7i2v3sz
+  subdir: ouroboros-consensus-shelley-test
```
Comment (mrBliss) [inline cabal.project:363]: Why is this needed?

Maybe you need to disable some `ouroboros-consensus*` testsuites in this file?

File: stack.yaml [hunk 4]
```diff
@@ -132,8 +132,10 @@ extra-deps:
         - ouroboros-consensus
         - ouroboros-consensus-byron
         - ouroboros-consensus/ouroboros-consensus-mock
+        - ouroboros-consensus/ouroboros-consensus-test-infra
         - ouroboros-consensus-cardano
         - ouroboros-consensus-shelley
+        - ouroboros-consensus-shelley-test
```
Comment (mrBliss) [inline stack.yaml:138]: Same remark as about `cabal.project`.
